x-pack/plugin/text-structure/src/main/java/org/elasticsearch/xpack/textstructure/structurefinder/TimestampFormatFinder.java

```diff
@@ -288,6 +289,43 @@ public final class TimestampFormatFinder {
         )
     );
 
+    /**
+     * The first match in this list will be chosen, so it needs to be ordered
+     * such that more generic patterns come after more specific patterns.
+     */
+    static final List<CandidateTimestampFormat> ORDERED_CANDIDATE_FORMATS_ECS_V1;
+    static {
+        List<CandidateTimestampFormat> items = new ArrayList<>();
+        // CATALINA8_DATESTAMP %{MONTHDAY}-%{MONTH}-%{YEAR} %{HOUR}:%{MINUTE}:%{SECOND}
+        // 29-Aug-2021 12:03:33.578
+        items.add(
+            new CandidateTimestampFormat(
+                example -> Collections.singletonList("dd-MMM-yyyy hh:mm:ss"),
+                "\\b\\d{2}-[A-Z]\\S{2}-\\d{4} \\d{2}:\\d{2}:\\d{2}[:.,]\\d{3}",
```

Comment:
If the comment is accurate and the separator is always `.` then this should be:

```suggestion
                example -> Collections.singletonList("dd-MMM-yyyy hh:mm:ss.SSS"),
                "\\b\\d{2}-[A-Z]\\S{2}-\\d{4} \\d{2}:\\d{2}:\\d{2}\\.\\d{3}",
```

Otherwise, if the fractional seconds separator really can vary between `.`, `,` and `:` even for this single type of logs then line 303 would need to be `example -> CandidateTimestampFormat.adjustFractionalSecondsFromEndOfExample("dd-MMM-yyyy hh:mm:ss"),` and line 304 can stay as-is.